tests/unit/test_charm.py

```diff
@@ -859,6 +860,74 @@ def test_given_magma_service_running_but_gateway_info_doesnt_return_values_for_s
             call("Failed to get Magma Access Gateway secrets!"),
         )
 
+    @patch("subprocess.check_output")
+    @patch("subprocess.run")
+    def test_given_magma_service_is_not_running_when_post_install_checks_action_then_checks_not_start(  # noqa: E501
+        self, patch_subprocess_run, patch_check_output
+    ):
+        patch_subprocess_run.return_value = Mock(returncode=1)
+        action_event = Mock()
+
+        self.harness.charm._on_post_install_checks_action(event=action_event)
+
+        patch_check_output.assert_not_called()
+        self.assertEqual(
+            action_event.fail.call_args,
+            call("Magma is not running! Please start Magma and try again."),
+        )
+
+    @patch("subprocess.run")
+    @patch("subprocess.check_output")
+    def test_given_magma_service_is_running_when_post_install_checks_action_then_checks_start(
+        self,
+        patch_subprocess_check_output,
+        patch_subprocess_run,
+    ):
+        action_event = Mock()
+        patch_subprocess_run.return_value = Mock(returncode=0)
+
+        self.harness.charm._on_post_install_checks_action(event=action_event)
+
+        patch_subprocess_check_output.assert_has_calls(
```

Comment:
For just one call you can use `assert_called_once_with`.

Reply:
Fixed.